### PR TITLE
Fix parameter load: Embedded parametersets

### PR DESCRIPTION
**Pull request by robinschmid**
getElementsByTagName is evil.
It goes through the whole tree not only the first level - this lead to loading of  SubParameterSet parameters on the parent parameter level

### CODE REVIEW

File: src/main/java/io/github/mzmine/main/impl/MZmineConfigurationImpl.java [hunk 1]
```diff
@@ -296,7 +295,13 @@ public void loadConfiguration(File file, boolean loadPreferences) throws IOExcep
...
 
           ParameterSet moduleParameters = getModuleParameters(moduleClass);
-          moduleParameters.loadValuesFromXML(moduleElement);
+          if(moduleParameters==null) {
+            logger.info(
+                "Module %s was in the config file but was not found in the current version of MZmine".formatted(
+                    moduleClass.getName()));
+          }
+          var parameterElement = (Element) moduleElement.getElementsByTagName("parameters").item(0);
+          moduleParameters.loadValuesFromXML(parameterElement);
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/main/impl/MZmineConfigurationImpl.java:304]: it can still be null here or?

**Review by SteffenHeu — COMMENTED** [review]
good catch

File: src/main/java/io/github/mzmine/main/impl/MZmineConfigurationImpl.java (hunk 1 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/main/impl/MZmineConfigurationImpl.java:304]: It actually can. But then it describes that in the exception

Comment (robinschmid) [reply src/main/java/io/github/mzmine/main/impl/MZmineConfigurationImpl.java:304]: in the try catch

**Pull request merged**